minio/credentials/providers.py

```diff
@@ -473,11 +474,17 @@ class StaticProvider(Provider):
     """Fixed credential provider."""
 
     def __init__(self, access_key, secret_key, session_token=None):
-        self._credentials = Credentials(access_key, secret_key, session_token)
+        self._access_key = access_key
+        self._secret_key = secret_key
+        self._session_token = session_token
```

Comment:
StaticProvider` should raise error right away for invalid access/secret keys.  It is meaningless accept invalid values here.

Reply:
Well, I don't see why it should behave differently from any other provider. Let's say we want to use EnvAWSProvider first, then AWSConfigProvider and only as a fallback the keys provided through the sw api, if any. I wouldn't prevent such application just to save a few lines of code.

Reply:
`StaticProvider` is a commonly used credential provider. This provider is expected to return credentials without any error. This behavior cannot be changed i.e. we shouldn't return exception on `retrieve()`.